tests/utils/Rule.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @class Rule
+ *
+ * @classdesc Simple get/set class to build a single Rule
+ */
+class Rule {
+    constructor() {
+        // defaults
+        this.id = 'test-id';
+        this.status = 'Enabled';
+        this.tags = [];
+    }
+
+    build() {
+        const rule = {};
+
+        rule.ID = this.id;
+        rule.Status = this.status;
+
+        if (this.expiration) {
+            rule.Expiration = this.expiration;
+        }
+        if (this.ncvExpiration) {
+            rule.NoncurrentVersionExpiration = this.ncvExpiration;
+        }
+        if (this.abortMPU) {
+            rule.AbortIncompleteMultipartUpload = this.abortMPU;
+        }
+
+        const filter = {};
+        if ((this.prefix && this.tags.length) || (this.tags.length > 1)) {
+            // And rule
+            const andRule = {};
+
+            if (this.prefix) {
+                andRule.Prefix = this.prefix;
+            }
+            andRule.Tags = this.tags;
+            filter.And = andRule;
+        } else {
+            if (this.prefix) {
+                filter.Prefix = this.prefix;
+            }
+            if (this.tags.length) {
+                filter.Tag = this.tags[0];
+            }
+        }
+        rule.Filter = filter;
+
+        return rule;
+    }
+
+    addID(id) {
```

Comment:
Minor, but maybe worth noting: I've seen setters/getters in this repo named with 'set', so instead this method would be `setID`.